Python/pytime.c

```diff
@@ -99,9 +99,13 @@ _PyTime_Round(double x, _PyTime_round_t round)
     else if (round == _PyTime_ROUND_CEILING) {
         d = ceil(d);
     }
-    else {
+    else if (round == _PyTime_ROUND_FLOOR) {
         d = floor(d);
     }
+    else {
+        assert(round == _PyTime_ROUND_UP);
+        d = (d < 0.0) ? floor(d) : ceil(d);
```

Comment:
Would you mind to write the positive case first please? :-)

(d >= 0.0) ? ceil(d) : floor(d)

Reply:
Fixed in a21d10d. :wink: